File: server_integration_tests/java/java_server_reactive/src/main/java/dev/graphlink/javaserver/impl/AuthorSchemaMappingsServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.interfaces.Article;
import dev.graphlink.javaserver.generated.services.AuthorSchemaMappingsService;
import dev.graphlink.javaserver.generated.types.Article;
import dev.graphlink.javaserver.generated.types.Author;
import graphql.GraphQLContext;

import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Service
public class AuthorSchemaMappingsServiceImpl implements AuthorSchemaMappingsService {

    @Override
    public Mono<Map<Author, List<Article>>> authorArticles(List<Author> value, GraphQLContext graphQLContext) {
        Map<Author, List<Article>> result = new HashMap<>();
        for (Author author : value) {
            List<Article> articles = Data.articles.stream()
                .filter(a -> a.getAuthorId().equals(author.getId()))
                .collect(Collectors.toList());
            result.put(author, articles);
        }
        return Mono.just(result);
    }

    @Override
    public Flux<Article> authorLatestArticles(Integer limit, Author value) {
        return Flux.fromIterable(Data.articles.stream()
            .filter(a -> a.getAuthorId().equals(value.getId()))
            .limit(limit)
            .collect(Collectors.toList()));
    }
}

File: server_integration_tests/servers/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/AuthorSchemaMappingsServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl


import dev.graphlink.kotlinserverblocking.generated.services.AuthorSchemaMappingsService
import dev.graphlink.kotlinserverblocking.generated.types.Article
import dev.graphlink.kotlinserverblocking.generated.types.Author
import org.springframework.stereotype.Service

@Service
class AuthorSchemaMappingsServiceImpl : AuthorSchemaMappingsService {

    override fun authorArticles(
        value: List<Author>
    ): Map<Author, List<Article>?> {
        val result = mutableMapOf<Author, List<Article>?>()
        for (author in value) {
            result[author] = Data.articles.filter { it.authorId == author.id }
        }
        return result
    }

    override fun authorLatestArticles(limit: Int, value: Author): List<Article> =
        Data.articles.filter { it.authorId == value.id }.take(limit)
}
